modules/wlroots/hm.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with config.stylix.colors;
+
+let
+  swwwUnits = serviceConfig: {
+    swww = {
+      Unit = {
+        Description = "Wallpaper daemon";
+        Documentation = [ "https://github.com/Horus645/swww#readme" ];
+        PartOf = [ "graphical-session.target" ];
+      };
+      Service = {
+        Type = "notify";
+        NotifyAccess = "all";
+        Environment = [ "PATH=${pkgs.swww}/bin" ];
+        ExecStart = "${pkgs.swww}/bin/swww init --no-daemon";
+        ExecStop = "${pkgs.swww}/bin/swww kill";
+      };
+    };
+
+    swww-wallpaper = {
+      Install = {
+        WantedBy = [ "graphical-session.target" ];
+      };
+      Unit = {
+        Description = "Apply wallpaper";
+        BindsTo = [ "swww.service" ];
+        After = [ "swww.service" ];
+        PartOf = [ "graphical-session.target" ];
+      };
+      Service = serviceConfig;
+    };
+  };
+
+  mpvUnits = file: {
+    mpv-wallpaper = {
+      Install = {
+        WantedBy = [ "graphical-session.target" ];
+      };
+      Unit = {
+        Description = "Video wallpaper player";
+        Documentation = [ "https://github.com/GhostNaN/mpvpaper#readme" ];
+        PartOf = [ "graphical-session.target" ];
+      };
+      Service.ExecStart = "${pkgs.mpvpaper}/bin/mpvpaper --auto-pause --mpv-options 'no-audio loop' '*' ${file}";
+    };
+  };
+
+  units = config.stylix.wallpaper.unpack rec {
+    image =
+      { file, ... }:
+      swwwUnits {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        ExecStart = "${pkgs.swww}/bin/swww img ${file}";
+        Environment = [ "SWWW_TRANSITION=none" ];
+      };
+
+    slideshow =
+      { files, delay, ... }:
+      let slideshow = pkgs.writeShellApplication {
+        name = "slideshow";
+        runtimeInputs = [ pkgs.swww ];
+        text = ''
+          images=(${builtins.toString files})
```

Comment:
This will fail for file names with spaces. Can probably be fixed using `lib.escapeShellArg`